cmake/python.cmake

```diff
@@ -10,32 +10,26 @@ message(STATUS "Using PYTHON_EXECUTABLE: ${PYTHON_EXECUTABLE}")
 
 set(_PYTHON_PATH_VERSION_SUFFIX "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}")
 
-set(enable_setuptools_deb_layout OFF)
-if(EXISTS "/etc/debian_version")
-  set(enable_setuptools_deb_layout ON)
+# setuptools is fussy about windows paths, make sure the install prefix is in native format
+if(WIN32)
+  file(TO_NATIVE_PATH "${CMAKE_INSTALL_PREFIX}" SETUPTOOLS_INSTALL_PREFIX)
 endif()
-option(SETUPTOOLS_DEB_LAYOUT "Enable debian style python package layout" ${enable_setuptools_deb_layout})
 
-if(SETUPTOOLS_DEB_LAYOUT)
-  message(STATUS "Using Debian Python package layout")
-  set(PYTHON_PACKAGES_DIR dist-packages)
-  set(SETUPTOOLS_ARG_EXTRA "--install-layout=deb")
-  # use major version only when installing 3.x with debian layout
-  if("${PYTHON_VERSION_MAJOR}" STREQUAL "3")
-    set(_PYTHON_PATH_VERSION_SUFFIX "${PYTHON_VERSION_MAJOR}")
+# this block determines the same value as the Python function get_python_install_dir() from python/catkin/builder.py
+if(NOT DEFINED PYTHON_INSTALL_DIR)
+  execute_process(COMMAND "${PYTHON_EXECUTABLE}"
```

Comment:
I think it would be easier for a tool to support this if the checking to make sure the python executables match is done in this cmake code, i.e. passing `PYTHON_EXECUTABLE:PYTHON_INSTALL_DIR` rather than just `PYTHON_INSTALL_DIR` from the tool. It is very difficult for the tool to determine if it has the same `PYTHON_EXECUTABLE` as this cmake code and since this CMake logic cannot be run separately (at least in its current configuration) the tool cannot do anything to resolve the mismatch if/when it was detected.

My suggestion would be to:

- separate the logic that calculates the `PYTHON_EXECUTABLE` and the `PYTHON_INSTALL_DIR` into a separate CMake file so it can be reused
  - this would allow the tool to make sure it is using the same `PYTHON_EXECUTABLE` and therefore `PYTHON_INSTALL_DIR` as the package will be using in most cases
    - this allows us to avoid the extra python invocation, without this I think it leaves a clear con for this pr (increased build time)
  - prevent the issue where the `PYTHONPATH` and installed subdirectory disagree
    - though I agree that this not the responsibility of this pr to solve, it is a nice benefit
- add some code here to handle the `PYTHON_EXECUTABLE:PYTHON_INSTALL_DIR` tuple input
  - protects the individual package from any unforeseen mismatches when the tool is attempting to precompute the `PYTHON_INSTALL_DIR` for the package
    - again this wouldn't be necessary if this pr didn't introduce an additional call to python, so I see it as in scope to provide a mechanism to mitigate that drawback

I can help implement the changes, but I think they're pretty important to consider and possibly get in before merging the other changes (which introduce the extra Python call).